DiverseSelector/test/test_selectors.py

```diff
@@ -120,7 +120,7 @@ def test_directedsphereexclusion():
     selector = DirectedSphereExclusion()
     selected_ids = selector.select(arr=coords_cluster, num_selected=12, labels=class_labels_cluster)
     # make sure all the selected indices are the same with expectation
-    assert_equal(selected_ids, [95, 14, 88, 84, 76, 68, 93, 50, 29, 19, 54])
+    assert_equal(selected_ids, [95, 14, 88, 84, 76, 49, 93, 50, 29, 27, 15, 54])
```

Comment:
Thanks for improving the codes!
It is noticed that the selected indices become different. Is this normal or do we have some explanations for this?

Reply:
I think kd-tree is only approximate in its fastest forms. The user should have the option to chose the brute-strength or kd-tree approach.

Reply:
Can you confirm that we have these two options? @alnaba1 Thanks.

Reply:
@PaulWAyers @FanwangM The change in selected indices was due to an oversight which has now been fixed. There is no need to have two options since Nearest Neighbor Search is exact (unlike the Furthest Neighbor Search that was done in the KDTree algorithm) and will always give the exact same result as a brute-strength option.